deep_quoridor/src/train_alphazero.py

```diff
@@ -64,117 +81,164 @@ def self_play_worker(
         print(f"Starting epoch {epoch}")
-        # Get the model state from our training agent so that it can be sent to all the worker
-        # processes and used for the self play games in each epoch.
-        model_state = training_agent.get_model_state()
 
         # Tell workers to run self play games for this epoch
         games_per_worker = int(np.ceil(args.games_per_epoch / len(workers)))
         games_remaining_to_allocate = args.games_per_epoch
         for worker in workers:
             this_worker_num_games = min(games_per_worker, games_remaining_to_allocate)
-            worker.job_queue.put(RunGamesJob(this_worker_num_games, self_play_params, model_state))
+            worker.params.job_queue.put(
+                RunGamesJob(this_worker_num_games, self_play_params, clear_evaluator_cache=True)
+            )
 
         # Collect results from the workers
+        results = []
         for worker in workers:
-            result = worker.result_queue.get()
+            results.append(worker.params.result_queue.get())
+
+        # Merge results into the primary replay buffer. We sort them to try and
+        # make the results more deterministic across multiple runs.
+        results = sorted(results, key=lambda r: r.worker_id)
+        for r in results:
             # NOTE: Make sure the replay buffer size for the training agent is large enough to hold
             # the replay buffer results from all agents each epoch or else we'll end up discarding
             # some results and we'll have wasted computation by playing those games.
-            training_agent.replay_buffer.extend(result.replay_buffer)
+            replay_buffer.extend(r.replay_buffer)
 
-        if len(training_agent.replay_buffer) > training_params.batch_size:
-            print(f"Training with {len(training_agent.replay_buffer)} turns in replay buffer")
-            metrics = training_agent.evaluator.train_iteration(training_agent.replay_buffer)
-            print(f"Training done: f{metrics}")
+        if len(replay_buffer) > training_params.batch_size:
+            print(f"Training with {len(replay_buffer)} turns in replay buffer")
+            metrics = evaluator_server.train_iteration(replay_buffer)
+            print(f"Training done: {metrics}")
         else:
             print(
-                f"Not training; batch_size={training_params.batch_size} is less than"
-                + f"replay buffer size ({len(training_agent.replay_buffer)})"
+                f"Not training; batch_size={training_params.batch_size} is less than "
+                + f"replay buffer size ({len(replay_buffer)})"
             )
 
+        print(evaluator_server.get_statistics())
+        for r in results:
+            print(r.evaluator_statistics)
+
+
+def setup(
+    board_size: int, max_walls: int, max_game_length: int, num_workers: int, random_seed: int
+) -> tuple[EvaluatorServer, list[Worker]]:
+    set_deterministic(random_seed)
+
+    set_deterministic(args.seed)
```

Comment:
You're setting it twice, I think it's the same value, but probably this one should be removed